test_scripts/API/VehicleData/commonVehicleData.lua

```diff
@@ -0,0 +1,289 @@
+          :Times(1)
+          self["mobileSession" .. id]:ExpectNotification("OnPermissionsChange"):Times(2)
+          EXPECT_HMICALL("VehicleInfo.GetVehicleData", { odometer = true})
+        end)
+    end)
+end
+
+function commonVehicleData.raiN(id, self)
+  self, id = commonVehicleData.getSelfAndParams(id, self)
+  if not id then id = 1 end
+  self["mobileSession" .. id] = mobile_session.MobileSession(self, self.mobileConnection)
+  self["mobileSession" .. id]:StartService(7)
+  :Do(function()
+      local corId = self["mobileSession" .. id]:SendRPC("RegisterAppInterface", config["application" .. id].registerAppInterfaceParams)
+      EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered", { application = { appName = config["application" .. id].registerAppInterfaceParams.appName } })
+      :Do(function(_, d1)
+          hmiAppIds[config["application" .. id].registerAppInterfaceParams.appID] = d1.params.application.appID
+        end)
+      self["mobileSession" .. id]:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+      :Do(function()
+          self["mobileSession" .. id]:ExpectNotification("OnHMIStatus", { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+          :Times(1)
+          self["mobileSession" .. id]:ExpectNotification("OnPermissionsChange")
+        end)
+    end)
+end
+
+local function allowSDL(self)
+  self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality",
+    { allowed = true, source = "GUI", device = { id = config.deviceMAC, name = "127.0.0.1" } })
+end
+
+function commonVehicleData.start(pHMIParams, self)
+  self, pHMIParams = commonVehicleData.getSelfAndParams(pHMIParams, self)
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self)
+  :Do(function()
+      self:initHMI(self)
+      :Do(function()
+          commonFunctions:userPrint(35, "HMI initialized")
+          self:initHMI_onReady(pHMIParams)
+          :Do(function()
+              commonFunctions:userPrint(35, "HMI is ready")
+              self:connectMobile()
+              :Do(function()
+                  commonFunctions:userPrint(35, "Mobile connected")
+                  allowSDL(self)
+                end)
+            end)
+        end)
+    end)
+end
+
+function commonVehicleData.unregisterApp(pAppId, self)
+  local mobSession = commonVehicleData.getMobileSession(self, pAppId)
+  local hmiAppId = commonVehicleData.getHMIAppId(pAppId)
+  local cid = mobSession:SendRPC("UnregisterAppInterface",{})
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { appID = hmiAppId, unexpectedDisconnect = false })
+  mobSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+end
+
+function commonVehicleData.backupHMICapabilities()
+  local hmiCapabilitiesFile = commonFunctions:read_parameter_from_smart_device_link_ini("HMICapabilities")
+  commonPreconditions:BackupFile(hmiCapabilitiesFile)
+end
+
+function commonVehicleData.restoreHMICapabilities()
```

Comment:
@AByzhynar Function is not used

Reply:
@dboltovskyi Fixed in 3b703abdab25b3e13b150954e8a8c707f40a3b45